ui/onboarding/Keycard/CreatePINModal.qml

```diff
@@ -0,0 +1,113 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Dialogs 1.3
+import StatusQ.Controls 0.1
+import "../../imports"
+import "../../shared"
+import "../../shared/keycard"
+
+ModalPopup {
+    property bool firstPINFieldValid: false
+    property bool repeatPINFieldValid: false
+    property string pinValidationError: ""
+    property string repeatPINValidationError: ""
+    property bool submitted: false
+
+    id: popup
+    title: qsTr("Create PIN")
+    height: 500
+
+    onOpened: {
+        submitted = false
+        firstPINField.text = "";
+        firstPINField.forceActiveFocus(Qt.MouseFocusReason)
+    }
+
+    Input {
+        id: firstPINField
+        anchors.rightMargin: 56
+        anchors.leftMargin: 56
+        anchors.top: parent.top
+        anchors.topMargin: 88
+        placeholderText:  qsTr("New PIN")
+        textField.echoMode: TextInput.Password
+        onTextChanged: {
+            [firstPINFieldValid, pinValidationError] =
+                Utils.validatePINs("first", firstPINField, repeatPINField);
+        }
+    }
+
+    Input {
+        id: repeatPINField
+        enabled: firstPINFieldValid
+        anchors.rightMargin: 0
+        anchors.leftMargin: 0
+        anchors.right: firstPINField.right
+        anchors.left: firstPINField.left
+        anchors.top: firstPINField.bottom
+        anchors.topMargin: Style.current.xlPadding
+        placeholderText: qsTr("Confirm PIN")
+        textField.echoMode: TextInput.Password
+        Keys.onReturnPressed: function(event) {
+            if (submitBtn.enabled) {
+                submitBtn.clicked(event)
+            }
+        }
+        onTextChanged: {
+            [repeatPINFieldValid, repeatPINValidationError] =
+                Utils.validatePINs("repeat", firstPINField, repeatPINField);
+        }
+    }
```

Comment:
Just an FYI:

These should ideally be `StatusInput`'s from `StatusQ, however, they don't support `echoMode` at the moment, so it's fine to use `Input` for now.